MapboxNavigation/StyleManager.swift

```diff
@@ -97,7 +97,24 @@ open class StyleManager: NSObject {
     }
     
     func applyStyle() {
-        guard let location = delegate?.locationFor(styleManager: self) else { return }
+        guard let location = delegate?.locationFor(styleManager: self) else {
+            // We can't calculate sunset or sunrise w/o a location so just apply all styles
+            for style in styles {
+                style.apply()
+                delegate?.styleManager?(self, didApply: style)
+            }
+            return
+        }
+        
+        // Single style usage
+        guard styles.count > 1 else {
```

Comment:
Is this block really needed? You could easily do the same thing below via:

```swift
for style in styles {
             if style.styleType == styleTypeForTimeOfDay || styles.count == 1 {
                 style.apply()
                 currentStyleType = style.styleType
                 delegate?.styleManager?(self, didApply: style)
             }
         }
```

Reply:
Yes less code is always nice, but IMO explicit guards make things a little more clear and they are easier to debug.